packages/odyssey-react-mui/src/Select.tsx

```diff
@@ -338,50 +321,52 @@ const Select = <
       const stopPropagation = (event: MouseEvent<SVGSVGElement>) =>
         event.stopPropagation();
 
+      const shouldRenderNonInteractiveIcon =
+        !isInteractive &&
+        controlledStateRef.current === CONTROLLED &&
+        hasMultipleChoices;
+
       return (
-        <ChipsInnerContainer
-          isInteractive={isInteractive}
-          odysseyDesignTokens={odysseyDesignTokens}
-        >
-          {selection.length <= 0 && (
-            <ChipsSpacer odysseyDesignTokens={odysseyDesignTokens} />
-          )}
-          {selection.map(
-            (item: string) =>
-              item?.length > 0 && (
-                <MuiChip
-                  key={item}
-                  label={
-                    <>
-                      {item}
-                      {!isInteractive &&
-                        controlledStateRef.current === CONTROLLED &&
-                        hasMultipleChoices && (
+        Array.isArray(internalSelectedValues) && (
+          <ChipsInnerContainer
+            isInteractive={isInteractive}
+            odysseyDesignTokens={odysseyDesignTokens}
+          >
+            {internalSelectedValues.map(
+              (item: string) =>
+                item?.length > 0 && (
```

Comment:
We don't want `item?.length` here if `item` is a string. And we shouldn't manually define the type for an array item because it's defined in the array.

If it can sometimes be empty, then `.filter(Boolean)` right before this `.map` will catch it.

Reply:
Should we rename `item` to `selectedValue`?